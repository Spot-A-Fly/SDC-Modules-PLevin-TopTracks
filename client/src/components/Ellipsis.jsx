import React, { Component } from 'react';

class Ellipsis extends Component {
  constructor() {
    super();
    this.state = {
      menu: false
    };
    this.toggleMenu = this.toggleMenu.bind(this);
  }

  toggleMenu() {
    const { menu } = this.state;
    this.setState({
      menu: !menu
    });
  }

  render() {
    const { menu } = this.state;
    switch (menu) {
      case false: return (
        <div className="ellipsis" onClick={this.toggleMenu}>
          <i className="fas fa-ellipsis-h" />
        </div>
      );
      case true: return (
        <div className="contextMenu" onClick={this.toggleMenu}>
          <div>Start Radio</div>
          <div>Save to Your Library</div>
          <div>Add to Queue</div>
          <div>Add to Playlist</div>
          <div>Copy Song Link</div>
        </div>
      );
      default: return (
        <div className="ellipsis">
          <i className="fas fa-ellipsis-h" />
        </div>
      );
    }
  }
}

export default Ellipsis;

// import React from 'react';
// import PropTypes from 'prop-types';

// function Ellipsis({ menu, showMenu }) {
//   switch (menu) {
//     case false: return (
//       <div className="ellipsis" onClick={showMenu}>
//         <i className="fas fa-ellipsis-h" />
//       </div>
//     );
//     case true: return (
//       <div className="contextMenu">
//         <div>Start Radio</div>
//         <div>Save to Your Library</div>
//         <div>Add to Queue</div>
//         <div>Add to Playlist</div>
//         <div>Copy Song Link</div>
//       </div>
//     );
//     default: return (
//       <div className="ellipsis">
//         <i className="fas fa-ellipsis-h" />
//       </div>
//     );
//   }
// }

// Ellipsis.propTypes = {
//   menu: PropTypes.isRequired,
//   showMenu: PropTypes.isRequired
// };
