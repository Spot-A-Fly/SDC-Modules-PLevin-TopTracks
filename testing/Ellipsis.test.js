import React from 'react';
import { render, cleanup, fireEvent } from 'react-testing-library';
import Ellipsis from '../client/src/components/Ellipsis.jsx';

afterEach(() => {
  cleanup();
  console.error.mockClear();
});

console.error = jest.fn();

test('<Ellipsis w/ ellipsis />', () => {
  const { getByTestId, container } = render(<Ellipsis />);
  expect(getByTestId('ellipsis').firstChild.tagName).toEqual('I');
  expect(console.error).not.toBeCalled();
  expect(container.firstChild).toMatchSnapshot();
});

test('<Ellipsis w/ context menu />', () => {
  const { getByTestId, container, queryByTestId } = render(<Ellipsis />);
  expect(queryByTestId('context-menu')).toBeFalsy();
  fireEvent.click(getByTestId('ellipsis'));
  expect(queryByTestId('ellipsis')).toBeFalsy();
  
  expect(getByTestId('context-menu').firstChild.textContent).toBe('Start Radio');
  expect(getByTestId('context-menu').childElementCount).toBe(5);
  expect(container.firstChild).toMatchSnapshot();

  fireEvent.click(getByTestId('context-menu'));
  expect(queryByTestId('ellipsis')).toBeTruthy();
});
