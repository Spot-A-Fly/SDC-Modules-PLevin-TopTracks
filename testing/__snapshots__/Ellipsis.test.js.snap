// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Ellipsis w/ context menu /> 1`] = `
<div
  class="contextMenu"
  data-testid="context-menu"
>
  <div>
    Start Radio
  </div>
  <div>
    Save to Your Library
  </div>
  <div>
    Add to Queue
  </div>
  <div>
    Add to Playlist
  </div>
  <div>
    Copy Song Link
  </div>
</div>
`;

exports[`<Ellipsis w/ ellipsis /> 1`] = `
<div
  class="ellipsis"
  data-testid="ellipsis"
>
  <i
    class="fas fa-ellipsis-h"
  />
</div>
`;
